library/src/image.cxx

```diff
@@ -290,6 +312,24 @@ void image::save(const std::string& path, SaveFormat format) const
   }
 }
 
+//----------------------------------------------------------------------------
+std::vector<unsigned char> image::saveBuffer(SaveFormat format) const
+{
+  switch (format)
+  {
+    case SaveFormat::PNG:
+      return this->Internals->SaveBuffer<vtkPNGWriter>();
+    case SaveFormat::JPG:
+      return this->Internals->SaveBuffer<vtkJPEGWriter>();
+    case SaveFormat::TIF:
+      throw write_exception("Cannot save to buffer in TIF format");
+    case SaveFormat::BMP:
+      return this->Internals->SaveBuffer<vtkBMPWriter>();
+  }
+
+  return {};
```

Comment:
Is there a c++ way to mark that last return as unreachable?
Would throwing an exception be more correct? even tho it will never happen as the switch is exhaustive

maybe having `case SaveFormat::PNG: default: return ...` as the last case would please codecov? it'd be kinda ugly tho :/